Add outer join BBE (#3331)
ballerina-platform/ballerina-distribution: examples/outer-join-clause/outer_join_clause.md

# Outer join clause

> Outer join clause perform `left outer equijoin`. The `outer join` clause include all of the records from the left table, even if there are no matching values for records in the right table. The type to join on must be `anydata`.

Reviewer: ```suggestion
the `outer join` clause performs a `left outer equijoin`. The `outer join` clause includes all of the records from the left table even if there are no matching values for records in the right table. The type to join must be `anydata`.
```
